qiskit/aqua/operators/list_ops/summed_op.py

```diff
@@ -57,18 +58,59 @@ def distributive(self) -> bool:
         return True
 
     def add(self, other: OperatorBase) -> OperatorBase:
+        """ Return Operator addition of self and other, overloaded by ``+``.
+
+        Note:
+            This appends other to ``self.oplist`` without checking other is already included or not.
+            If you want to simplify this, please use ``SummedOp.simplify``.
+
+        Args:
+            other: An ``OperatorBase`` with the same number of qubits as self, and in the same
+                'Operator', 'State function', or 'Measurement' category as self (i.e. the same type
+                of underlying function).
+
+        Returns:
+            A ``SummedOp`` equivalent to the sum of self and other.
+        """
         if self == other:
             return self.mul(2.0)
-        elif isinstance(other, SummedOp):
-            self_new_ops = [op.mul(self.coeff) for op in self.oplist]
-            other_new_ops = [op.mul(other.coeff) for op in other.oplist]
-            return SummedOp(self_new_ops + other_new_ops)
-        elif other in self.oplist:
-            new_oplist = copy.copy(self.oplist)
-            other_index = self.oplist.index(other)
-            new_oplist[other_index] = new_oplist[other_index] + other
-            return SummedOp(new_oplist, coeff=self.coeff)
-        return SummedOp(self.oplist + [other], coeff=self.coeff)
+
+        self_new_ops = self.oplist if self.coeff == 1 \
+            else [op.mul(self.coeff) for op in self.oplist]
+        if isinstance(other, SummedOp):
+            other_new_ops = other.oplist if other.coeff == 1 \
+                else [op.mul(other.coeff) for op in other.oplist]
+        else:
+            other_new_ops = [other]
+        return SummedOp(self_new_ops + other_new_ops)
+
+    def simplify(self) -> 'SummedOp':
+        """ Return Operator by simplifying duplicate operators,
+            e.g., (2 * X ^ Y) + (X ^ Y) -> (3 * X ^ Y).
```

Comment:
The first line of a docstring should not include any linebreaks (also no whitespace between `"""` and the text, see https://ibm.ent.box.com/file/597503488790)
```suggestion
        """Return a simplified operator by summarizing duplicate terms.
        
        E.g. ``(2 * X ^ Y) + (X ^ Y)`` simplifies to ``(3 * X ^ Y)``.
```

Reply:
Thanks. I fix it.

Reply:
Some modules such as `ListOp` does not follow the rule. It would be good to deal with it in another PR.